cli/src/commands/log.rs

```diff
@@ -76,27 +76,53 @@ pub(crate) fn cmd_log(
 ) -> Result<(), CommandError> {
     let workspace_command = command.workspace_helper(ui)?;
 
-    let revset_expression = {
-        let mut expression = if args.revisions.is_empty() {
+    ui.request_pager();
+    let revset = {
+        let expression = if args.revisions.is_empty() {
             workspace_command.parse_revset(&command.settings().default_revset())?
         } else {
             workspace_command.parse_union_revsets(&args.revisions)?
         };
-        if !args.paths.is_empty() {
+        if args.paths.is_empty() {
+            workspace_command.evaluate_revset(expression)?
+        } else {
             let repo_paths: Vec<_> = args
                 .paths
                 .iter()
                 .map(|path_arg| workspace_command.parse_file_path(path_arg))
                 .try_collect()?;
-            expression = expression.intersection(&RevsetExpression::filter(
-                RevsetFilterPredicate::File(Some(repo_paths)),
-            ));
+            let path_filter =
+                &RevsetExpression::filter(RevsetFilterPredicate::File(Some(repo_paths)));
+
+            let revset = workspace_command.evaluate_revset(expression.intersection(path_filter))?;
+            if !revset.is_empty() {
```

Comment:
Perhaps, this kind of stuff can be done at the bottom to minimize the overhead.

Reply:
If we want to automatically drop the revset and only filter for the path, then this needs to be at the top. Also it's only evaluating a second revset if both path and revset are specified and if the intersection of that is empty with the latest change